app/controllers/api/v1/workflows_controller.rb

```diff
@@ -34,14 +34,11 @@ def index
       end
 
       def destroy
-        workflow = Workflow.find(params.require(:id))
-        workflow.destroy!
+        WorkflowDeleteService.new(params.require(:id)).delete
         head :no_content
       rescue ActiveRecord::InvalidForeignKey => e
         json_response({ :message => e.message }, :forbidden)
```

Comment:
@lgalis suggests to improve the error message when the workflow still has requests link to it.
Maybe we should explicitly detect the association and not rely on the rails build-in protection. The logic should be placed in `WorkflowDeleteService`.